server/src/main/webapp/WEB-INF/rails/webpack/models/pipeline_configs/pipeline_config.ts

```diff
@@ -281,4 +281,20 @@ export class Materials extends Array<Material> {
       this.splice(index, 1);
     }
   }
+
+  scmMaterialsHaveDestination(): boolean {
+    const hasScmMaterialWithEmptyDestination          = this
+      .filter((material) => material.attributes() instanceof ScmMaterialAttributes)
+      .some((material) => {
+        const attrs = material.attributes() as ScmMaterialAttributes;
+        return _.isEmpty(attrs.destination());
+      });
```

Comment:
this would iterate the collection twice.
How about moving the `material.attributes() instanceof ScmMaterialAttributes` check inside `some`?

Reply:
IMO, the iteration over the collection twice is the worst case scenario. Looping over collection twice is not necessary a performance issue. I factored readability before it.

Reply:
Sure, everyone will have their own preferences of writing code. :)

In this particular scenario, we are iterating over the collection 4 times in worst case. Best case, still has to do 2 iterations. Where, everything can be achieved in just 1 iteration.